src/lesson9/taskNumber1/QA27_9_1_Lugovskiy.java

```diff
@@ -0,0 +1,74 @@
+package lesson9.taskNumber1;
+
+import java.io.IOException;
+import java.util.InputMismatchException;
+import java.util.Random;
+import java.util.Scanner;
+
+public class QA27_9_1_Lugovskiy{
+    public static void main(String[] args) {
+        System.out.print("Enter the number of animals : ");
+        Animal[] animal = new Animal[0];
+        while (true) {
+            Scanner scanner1 = new Scanner(System.in);
+            try {
+                animal = new Animal[scanner1.nextInt()];
+            } catch (NegativeArraySizeException e) {
+                System.out.println("Array length cannot be negative");
+                continue;
+            } catch (InputMismatchException e) {
+                System.out.println("The length of the array must have a numeric expression");
+                continue;
+            }
+                break;
+        }
+        for (int i = 0; i < animal.length; i++) {
+            animal[i] = getNextAnimal();
+        }
+        System.out.println("Enter an action for animals to perform");
+        int choise;
+        while (true) {
+            System.out.println("0 - walk, 1 - talk, 2 - jump, 3 - eat, 4 - swim");
+            Scanner scanner = new Scanner(System.in);
+            try {
+                choise = scanner.nextInt();
+            } catch (InputMismatchException e) {
+                System.out.println("Action is only a number, not a symbol");
+                continue;
+            }
+            if (choise < 0 || choise > 4) {
+                try {
+                    throw new IOException();
+                }catch (IOException e){
```

Comment:
Strange construction. We don't need to use constants to prevent out of bounds error or to use exeption to catch it in the next line. Check the last lection, it has an example